test/ipns.js

```diff
@@ -0,0 +1,328 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const series = require('async/series')
+const parallel = require('async/parallel')
+const os = require('os')
+const path = require('path')
+const hat = require('hat')
+
+const DaemonFactory = require('ipfsd-ctl')
+const goDf = DaemonFactory.create()
+const jsDf = DaemonFactory.create({ type: 'js' })
+
+describe('ipns', () => {
+  it('should publish an ipns record to a js daemon and resolve it using the same js daemon', function (done) {
+    this.timeout(100 * 1000)
+
+    const dir = path.join(os.tmpdir(), hat())
+    const ipfsRef = '/ipfs/QmPFVLPmp9zv5Z5KUqLhe2EivAGccQW2r7M7jhVJGLZoZU'
+
+    let jsDaemon
+    let jsId
+
+    console.log('ipfsRepo', dir)
+
+    series([
+      (cb) => jsDf.spawn({
+        repoPath: dir,
+        disposable: false,
+        initOptions: { bits: 512 }
+      }, (err, node) => {
+        expect(err).to.not.exist()
+        jsDaemon = node
+        jsDaemon.init(cb)
+      }),
+      (cb) => jsDaemon.start(cb),
+      (cb) => jsDaemon.api.id((err, res) => {
+        expect(err).to.not.exist()
+        jsId = res.id
+        cb()
+      }),
+      (cb) => jsDaemon.api.name.publish(ipfsRef, { resolve: false }, cb),
+      (cb) => {
+        jsDaemon.api.name.resolve(jsId, (err, res) => {
+          expect(err).to.not.exist()
+          expect(res).to.equal(ipfsRef)
+          cb()
+        })
+      },
+      (cb) => jsDaemon.stop(cb)
+      // (cb) => setTimeout(cb, 10500)
+    ], done)
+  })
+
+  it('should publish an ipns record to a go daemon and resolve it using the same go daemon', function (done) {
```

Comment:
I am having strange problems with this test. I need to have two go daemons running, in order to be able to publish an IPNS record. Otherwise, I receive `Error: failed to find any peer in table`